CRM/Member/BAO/Membership.php

```diff
@@ -2129,7 +2129,12 @@ protected static function matchesRequiredMembership($params, $membership) {
    * @throws \CRM_Core_Exception
    */
   protected static function hasExistingInheritedMembership($params) {
-    foreach (civicrm_api3('Membership', 'get', ['contact_id' => $params['contact_id']])['values'] as $membership) {
+    $currentInheritedMemberships = \Civi\Api4\Membership::get(FALSE)
+      ->addJoin('MembershipStatus AS membership_status', 'LEFT')
+      ->addWhere('membership_status.is_current_member', '=', TRUE)
```

Comment:
This is the only real change - we filter by `is_current_member`

Reply:
and what about people that have a current membership status will this be covered :-P

Reply:
I originally named the variable `$currentInheritedMemberships` but it's actually `$currentMemberships` so I've changed it.
We could probably simplify further and do more work in the API call but this keeps the logic the same.